### ISSUE TITLE
Unexpected error on Grafana Cloud

### ISSUE THREAD

**Issue by lublot** [grafana/grafana-amazonprometheus-datasource#195]
Hello everyone! Thank you for developing this plugin, it certainly will be helpful to community.

Today I decided to install this plugin on Grafana Cloud and I got some unexpected error. After some investigation, I believe I found the problem in this piece of code.

https://github.com/grafana/grafana-amazonprometheus-datasource/blob/36fda9269f1d209c9cdcd8b23b8dd97defe631a6/src/configuration/DataSourceHttpSettingsOverhaul.tsx#L29

The short error stack is:

```
An unexpected error happened
View more
TypeError: Cannot assign to read only property 'sigV4Auth' of object '#<Object>'

at ZT (https://xxxxxx.grafana.net/public/plugins/grafana-amazonprometheus-datasource/module.js?_cache=1.0.0:2:681022)
at https://xxxxxx.grafana.net/public/plugins/grafana-amazonprometheus-datasource/module.js?_cache=1.0.0:2:683505
at div
at Ss (https://grafana-assets.grafana.net/grafana/11.2.0-73451/public/build/Connections.e81c03793a15227343bb.js:48:2378)
at r (https://grafana-assets.grafana.net/grafana/11.2.0-73451/public/build/6311.8c8758f3a48b051a272e.js:1:183)
at form
at Es (https://grafana-assets.grafana.net/grafana/11.2.0-73451/public/build/Connections.e81c03793a15227343bb.js:48:5847)
at Is (https://grafana-assets.grafana.net/grafana/11.2.0-73451/public/build/Connections.e81c03793a15227343bb.js:48:5404)
at p (https://grafana-assets.grafana.net/grafana/11.2.0-73451/public/build/6311.8c8758f3a48b051a272e.js:228:17993)
at div
at div
at div
at u (https://grafana-assets.grafana.net/grafana/11.2.0-73451/public/build/6311.8c8758f3a48b051a272e.js:228:17324)
at div
at K (https://grafana-assets.grafana.net/grafana/11.2.0-73451/public/build/6311.8c8758f3a48b051a272e.js:228:21917)
at $s (https://grafana-assets.grafana.net/grafana/11.2.0-
```

Could someone provide me a help about how to solve it?

**Comment by sdg9670** [comment]
I got same error

```
An unexpected error happened
Details
TypeError: Cannot assign to read only property 'sigV4Auth' of object '[object Object]'

    at ZT (https://test.grafana.net/public/plugins/grafana-amazonprometheus-datasource/module.js?_cache=1.0.0:2:681022)
    at https://test.grafana.net/public/plugins/grafana-amazonprometheus-datasource/module.js?_cache=1.0.0:2:683505
    at div
    at Ss (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/Connections.f0d3fc3c0875f139e040.js:48:2378)
    at r (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:1:183)
    at form
    at Es (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/Connections.f0d3fc3c0875f139e040.js:48:5847)
    at Is (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/Connections.f0d3fc3c0875f139e040.js:48:5404)
    at p (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:231:17991)
    at div
    at div
    at div
    at d (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:231:17322)
    at div
    at K (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:231:21920)
    at $s (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/Connections.f0d3fc3c0875f139e040.js:48:8783)
    at Os (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/Connections.f0d3fc3c0875f139e040.js:48:9179)
    at V (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1018:29174)
    at V (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1018:31352)
    at Js (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/Connections.f0d3fc3c0875f139e040.js:56:2221)
    at Suspense
    at g (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:63:133851)
    at Tr (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:187:12305)
    at V (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1018:29174)
    at lo (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1009:4452)
    at as (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1009:12001)
    at Ut (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1018:17388)
    at V (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1018:31352)
    at div
    at https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:82:52949
    at main
    at div
    at div
    at div
    at Cs (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:187:2406)
    at div
    at bi (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:187:7580)
    at lo (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1009:4452)
    at as (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1009:12001)
    at fs (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1009:11337)
    at yr (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1018:17687)
    at V (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1018:25388)
    at v (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:164:8978)
    at b (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1371:112899)
    at c (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:432:27874)
    at g (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:63:133851)
    at o (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:63:134360)
    at wt (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/9006.b4cc3b7e5a15e74b7f70.js:1375:508)
    at qi (https://grafana-assets.grafana.net/grafana/11.2.0-73830/public/build/6311.e2995c9b9e3ad85768d7.js:187:14420)
```

**Comment by bohandley** [comment]
One fix is to revert this -> https://github.com/grafana/grafana-amazonprometheus-datasource/pull/191/files

**Comment by njvrzm** [comment]
True, although I worry we'll get people using the datasource with non-Amazon authentication and then have to figure out how to migrate _them_ later.

**Comment by njvrzm** [comment]
This is fixed now with version 1.0.1 of the plugin - once you update your plugin version you should be able to proceed. Please let us know if you run into any more problems.

**Comment by sdg9670** [comment]
So thanks, I will check soon!
And let you know🙂
